src/docs/implementations/otlp.adoc

```diff
@@ -0,0 +1,131 @@
+----
+
+`OtlpConfig` is an interface with a set of default methods. If, in the implementation of `get(String k)`, rather than returning `null`, you instead bind it to a property source, you can override the default configuration through properties. For example, Micrometer's Spring Boot support binds properties prefixed with `management.otlp.metrics.export` directly to the `OtlpConfig`:
+
+[source, yaml]
+----
+management:
+  otlp:
+    metrics:
+      export:
+        # Supported configs
+        url: "https://otlp.example.com:4318/v1/metrics"
+        aggregationTemporality: "cumulative"
+        headers: "header1=value1"
+        resourceAttributes: "key=value"
+----
+
+1. url - The URL to which data will be reported. Defaults to `http://localhost:4318/v1/metrics`
+2. aggregationTemporality - https://opentelemetry.io/docs/specs/otel/metrics/data-model/#temporality[Aggregation temporality] determines how the additive quantities are expressed, in relation to time. The supported values are `cumulative` or `delta`. Defaults to `cumulative`. +
+*Note*: This config is introduced in version 1.11.0.
+3. headers - Additional headers to send with exported metrics. This can be used for authorization headers. By default, headers will be loaded from the config. If that is not set, they will be taken from the environment variables `OTEL_EXPORTER_OTLP_HEADERS` and `OTEL_EXPORTER_OTLP_METRICS_HEADERS`. If a header is set in both the environmental variables, the header in the latter will overwrite the former.
+4. resourceAttributes - https://opentelemetry.io/docs/specs/otel/resource/semantic_conventions/#service[Resource attributes] that will be used for all metrics published. By default micrometer adds the following resources,
+[%autowidth]
+|===
+|Key | Default value
+
+|telemetry.sdk.name
+|io.micrometer
+
+|telemetry.sdk.language
+|java
+
+|telemetry.sdk.version
+|<micrometer-core-version> (E.g: 1.11.0)
+
+|service.name
+|unknown_service
+|===
+
+If this config is empty, then resource attributes will be loaded from the environmental variable `OTEL_RESOURCE_ATTRIBUTES`. `service.name` can be overridden by the environmental variable `OTEL_SERVICE_NAME` and this takes precedence over other configs.
+
+== Supported metrics
+https://opentelemetry.io/docs/specs/otel/metrics/data-model/#metric-points[Metric points] define the different data-points that are supported in OTLP. Micrometer supports exporting the below data-points in OTLP format,
+
+1. https://opentelemetry.io/docs/specs/otel/metrics/data-model/#sums[Sums]
+2. https://opentelemetry.io/docs/specs/otel/metrics/data-model/#gauge[Gauge]
+3. https://opentelemetry.io/docs/specs/otel/metrics/data-model/#histogram[Histogram]
+4. https://opentelemetry.io/docs/specs/otel/metrics/data-model/#summary-legacy[Summary]
+
+The below table maps OTLP datapoint and the micrometer meters,
+[%autowidth]
+|===
+|OTLP DataPoint | Micrometer meter type
```

Comment:
I'm thinking if it would make more sense to do the mapping the other way Micrometer -> OTLP and also add Meter, TimeGauge, MultiGauge.

Reply:
@lenin-jaganathan GH somewhat hides this but what do you think?

Reply:
I initially started with that approach and I saw that table grew big, so I kind of moved away from that. Let me re-visit that. Also, will it be fine if we do N-1 mapping for micrometer meters or do you think 1-1 is needed?

Reply:
N-1 is completely good, please feel free to keep it as is if it makes more sense to you, but could you please add `Meter`, `TimeGauge`, `MultiGauge` too? I guess all of them translate to gauge in OTLP.

Reply:
In the case of Otlp, I guess MultiGauge is not implemented (same as an arbitrary Meter). Also, currently, we are not exporting anything for the `Meter`. Do you agree with the 2 statements?

I haven't verified the behaviour but by looking at the code, it seems so.

Reply:
MultiGauge is a bit of a misnomer in that it itself is not a Meter, but you use its API to register gauges.
Meter will I think publish the metadata about the meter but without any datapoints. We still need to implement that.